### PR TITLE
Fix parsing empty strings

### PR DESCRIPTION
**Pull request by zhongzhi107**
This PR fixes a bug when using empty strings on update with the `patch` option. Since the values aren't being stringified, an error was raised when parsing them.

### CODE REVIEW

**Comment by ricardogama** [comment]
Thanks @zhongzhi107 for the contribution.

However, this doesn't seem to fix #49, as @pandapaul explained the error is thrown when a value was already stringified on save, or is malformed in the database and it fails when parsing on fetch.

Also, you seem to be fixing `null` or `undefined` values and I think we already cover that on the test suite. If I'm wrong can you please add tests to this use case?

**Comment by zhongzhi107** [comment]
Hi @ricardogama , I add test case.
An error occurs when updating an field with an empty string. 

`SyntaxError: Unexpected end of JSON input
      at JSON.parse (<anonymous>)
      at Object.keys.forEach.attribute (/home/q/www/qrmaps_api.qunar.com/node_modules/bookshelf-json-columns/dist/index.js:114:48)
      at Array.forEach (native)
      at Draft.Model.save.call.then.model (/home/q/www/qrmaps_api.qunar.com/node_modules/bookshelf-json-columns/dist/index.js:112:33)
      at Draft.tryCatcher (/home/q/www/qrmaps_api.qunar.com/node_modules/bluebird/js/release/util.js:16:23)
      at Promise._settlePromiseFromHandler (/home/q/www/qrmaps_api.qunar.com/node_modules/bluebird/js/release/promise.js:512:31)
      at Promise._settlePromise (/home/q/www/qrmaps_api.qunar.com/node_modules/bluebird/js/release/promise.js:569:18)
      at Promise._settlePromise0 (/home/q/www/qrmaps_api.qunar.com/node_modules/bluebird/js/release/promise.js:614:10)
      at Promise._settlePromises (/home/q/www/qrmaps_api.qunar.com/node_modules/bluebird/js/release/promise.js:693:18)
      at Async._drainQueue (/home/q/www/qrmaps_api.qunar.com/node_modules/bluebird/js/release/async.js:133:16)
      at Async._drainQueues (/home/q/www/qrmaps_api.qunar.com/node_modules/bluebird/js/release/async.js:143:10)
      at Immediate.Async.drainQueues (/home/q/www/qrmaps_api.qunar.com/node_modules/bluebird/js/release/async.js:17:14)`

File: src/index.js [hunk 1]
```diff
@@ -103,7 +103,9 @@ export default Bookshelf => {
           // Parse JSON columns.
           Object.keys(attributes).forEach(attribute => {
             if (this.constructor.jsonColumns.includes(attribute)) {
-              model.attributes[attribute] = JSON.parse(model.attributes[attribute]);
+              if (model.attributes[attribute]) {
```
Comment (ricardogama) [inline src/index.js]: Please update to the following to keep consistency with other conditionals like this:

```js
if (this.constructor.jsonColumns.includes(attribute) && attributes[attribute]) {
  model.attributes[attribute] = JSON.parse(model.attributes[attribute]);
}
```

File: test/mysql/index.js [hunk 2]
```diff
@@ -140,6 +140,19 @@ describe('with MySQL client', () => {
       sinon.restore(ModelPrototype);
     });
 
+    it('should not stringify \'\' values on update with `patch` option', async () => {
```
Comment (ricardogama) [inline test/mysql/index.js]: This test does not assert that the value has not been parsed, change to the following:

```js
it('should keep an empty string on update with `patch` option', async () => {
  const model = await Model.forge({ foo: 'bar' }).save();

  await model.save({ foo: '' }, { patch: true });

  model.get('foo').should.equal('');
});
```

Comment (ricardogama) [inline test/mysql/index.js]: Also, please add the same test to the SQLite3 suite.

**Comment by zhongzhi107** [comment]
Thanks @ricardogama for the advices. 
The code have been modified and please review it again.

File: dist/index.js [hunk 3]
```diff
@@ -111,7 +111,9 @@ exports.default = Bookshelf => {
         // Parse JSON columns.
         Object.keys(attributes).forEach(attribute => {
           if (this.constructor.jsonColumns.indexOf(attribute) !== -1) {
-            model.attributes[attribute] = JSON.parse(model.attributes[attribute]);
```
Comment (ricardogama) [inline dist/index.js]: All good, but please remove changes on this file as the `dist` files are only added on the release commit.

**Comment by zhongzhi107** [comment]
I'm sorry for my carelessness. The changes have been discarded.

**Comment by ricardogama** [comment]
Thanks @zhongzhi107, released as [2.1.1](https://github.com/seegno/bookshelf-json-columns/releases/tag/2.1.1).

**Pull request merged**